src/te/operation/compute_op.cc

```diff
@@ -57,8 +57,10 @@ TVM_REGISTER_NODE_TYPE(ComputeOpNode);
 static void VerifyComputeOp(const ComputeOpNode* op);
 
 inline bool ReduceEqual(const tir::ReduceNode* a, const tir::ReduceNode* b) {
+  StructuralEqual expr_equal;
   return (a->combiner.same_as(b->combiner)) && (a->source.same_as(b->source)) &&
-         (a->axis.same_as(b->axis)) && (a->condition.same_as(b->condition)) &&
+         (a->axis.same_as(b->axis)) &&
+         (a->condition.same_as(b->condition) || expr_equal(a->condition, b->condition)) &&
```

Comment:
expr_equal(a->condition, b->condition) should can cover a->condition.same_as(b->condition), condition.same_as seems like not necessary.